rio_tiler/tasks.py

```diff
@@ -36,12 +46,15 @@ def filter_tasks(
     if not allowed_exceptions:
         allowed_exceptions = ()
 
-    for future, asset in tasks:
+    while True:
         try:
+            future, asset = next(tasks)  # type: ignore
             if isinstance(future, futures.Future):
                 yield future.result(), asset
             else:
                 yield future, asset
+        except StopIteration:
```

Comment:
we put StopIteration first, so someone doesn't try to put StopIteration as `allowed_exceptions`

Reply:
They might wonder why their endless loop is slow...

Reply:
well if they do that (putting StopIteration in allowed_exception) the while loop while never exit ...

Reply:
haha yes, hence the endless loop will be slow 😄